papas/pfalgo/blockbuilder.py

```diff
@@ -22,36 +20,29 @@ class BlockBuilder(GraphBuilder):
         history_nodes : dictionary of nodes that describe which elements are parents of which blocks 
                         if a history_nodes tree is passed in then 
                         the additional history will be added into the exisiting history 
-        pfevent : the particle flow event object which is needed so that the underlying object can 
-                be retrieved
         nodes : a set of nodes corresponding to the unique ids which is used to construct a graph
                 and thus find distinct blocks
         blocks: the resulting blocks
     
         
         Usage example:
 
-            builder = BlockBuilder(ids, edges, history_nodes, pfevent)
+            builder = BlockBuilder(ids, edges, history)
```

Comment:
At the moment we have the following
Graphbuilder = Uses edges (is_linked, distance) to create disconnected subgraphs
Blockbuilder(graphbuilder) = Takes ids and edges uses graphbuilder to make subgraphs and then creates PFBlocks from the subgraphs and adds blocks into the history
PFBlockBuilder(Blockbuilder) = Creates edge information and then uses Blockbuilder
PFBlockSplitter(Blockbuilder) = Gets edge information and then uses Blockbuilder
mergedclusterbuilder.py(graphbuilder) takes clusters and distance, creates edges and then uses graphbuilder and then makes merged clusters from subgraphs

thoughts:
1. PFBlockSplitter is now handled more simply and could perhaps dissappear altogether. This would be BlockBuilder and PFBlockBuilder could be merged.
2. Maybe these should have graphbuilder/blockbuilder as a component rather than use inheritance??